### ISSUE TITLE
One Iteration is taking 5 minutes in Training detector of logo

### ISSUE THREAD

**Issue by vg123** [AlexeyAB/darknet#89]
I am trying to train a detector for logo detection of car image and also classify it simultaneously using `yolo` and  `darknet` framework. 
I followed this [link](https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects) and tried to built a detector for my dataset and I just keep 6 image files.
I followed the instructions ::
created a file `yolo-obj.cfg` by copying from `yolo-v0c.2.0.cfg` with changes in lines 
```
2) batch=64
3) subdivisions=8
224) filters=35
230) classes=2
```
I created files in `data` folder 
`obj.names`
`obj.data`
`train.txt` 
`test.txt`

and created `obj` folder in `/darknet/data`
`/darknet/data/obj` includes my dataset of images with corresponding `.text`files.

My problem is that it is taking 5 or more minutes for each iteration to complete in training and also not generating any `.weight` file in `backup` folder for these iterations. and we are suppose to let it train for more than 7000 iterations. 

I used this command to train model `./darknet detector train data/obj.data yolo-obj.cfg darknet19_448.conv.23`

And I also tried it for multiple processor, firstly by setting `GPU=1` in MAKEFILE  (`darknet/MAKEFILE`) , than run command in terminal `make` and than train it with command  `./darknet detector train data/obj.data yolo-obj.cfg darknet19_448.conv.23 -gpus 0,1,2,3` but now also it is taking same time.

I was thinking that after setting `GPU=1` and appending `-gpus 0,1,2,3` in command as shown above, this training will occupy all 4 processors, but it didn't happen. Suggest me a way of to make it able to run on all processors and help me, by giving some method for training my model fast and affectively.

**Comment by AlexeyAB** [comment]
* How many nVidia GPU video cards are in your computer?
* What GPUs do you use?
* What version of CUDA and cuDNN do you use?
* Is there directory `/darknet/backup/`?
* Are any `*.weights` files in the directory `/darknet/backup/` after 7000 iterations of training?

* You should have at least 100 images per 1 class, but preferable 2000 images per class in your dataset for training. I.e. for your 2 classes you should have 200 - 4000 images for training.

**Comment by vg123** [comment]
Thank you for quick response,
I don't have GPU machine but it has 4 cores and i can also manage 16 cores.
I don't use any gpus I just made those changes in the files, and i donot have CUDA and cuDNN in my system.
Yes I have `backup`  directory, and there were some `weight` files which were created when I was training model using `CIFAR-10` dataset , but I had moved all those `weight` files some where else and now its empty folder.
And as you are saying to keep dataset size more than 200 per class but the process of training needs the `.txt` file along with image file which need much calculations and I did it manually which took lots of time so tell me some other way of finding these '<x><y><width><height>' values which needs to be put in `.txt` file corresponding to each image file.

**Comment by AlexeyAB** [comment]
Training on CPU will take about ~1-10 **years**.
You should use GPU to train Yolo, and it will take ~20 hours, depends on GPU-model.

To create .txt-files for new images you should use Yolo-mark on Win/Linux: https://github.com/AlexeyAB/Yolo_mark

**Comment by vg123** [comment]
what if I use more number of cores to train it (ie. 16 cores). I know it is repeating but please answer it

**Comment by AlexeyAB** [comment]
Yolo can use only 1 CPU core, and any number of GPUs (recomended not more than 4).

So, you can't use more than 1 CPU-Core to train Yolo: https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L133


Only for GPU used `train_networks` instead of `train_network`
```
#ifdef GPU
        if(ngpus == 1){
            loss = train_network(net, train);
        } else {
            loss = train_networks(nets, ngpus, train, 4);
        }
#else
        loss = train_network(net, train);
#endif
```

**Comment by vg123** [comment]
Actually I am planning to train my model for 40 different classes and I want to take just 20 images per class for training than how many hours, it will take for completion of training, if I use cpu only.

And why it is so that I needs to iterate for minimum 1000 iterations for generation of `weight` file, how can I reduce it and what will be the effect of it on trained model.

**Comment by AlexeyAB** [comment]
@vg123 If you want to train model for 40 different classes then you should have 80 000 images and do 80 000 iterations. Otherwise, the accuracy will be noticeably worse.

Yes, you can use EC2 instance on AWS, then you should install CUDA 8.0 and cuDNN v5.1 (for CUDA 8.0), and set `GPU=1` and `CUDNN=1` in the `Makefile`.
p2.xlarge - NVIDIA Tesla K80 (2 x GK210): https://aws.amazon.com/ec2/instance-types/

This video-card has two separate GPUs (2 x GK210), so fo fast training, you can use multi-GPU training after first 1000 iterations: https://github.com/AlexeyAB/darknet#how-to-train-with-multi-gpu

----

Also about this: https://github.com/AlexeyAB/darknet/issues/8

**Comment by vg123** [comment]
How many iterations will be required if I give 100 images per class.

**Comment by AlexeyAB** [comment]
@vg123 You need the same 2000 iterations per class, regardless of how much you have pictures.
